genie-agent/src/main/java/com/netflix/genie/agent/execution/process/impl/JobProcessManagerImpl.java

```diff
@@ -255,8 +258,15 @@ public JobProcessResult waitFor() throws InterruptedException {
                     return new JobProcessResult
                         .Builder(JobStatus.KILLED, JobStatusMessages.JOB_MARKED_FAILED, exitCode)
                         .build();
-                case API_KILL_REQUEST:
                 case SYSTEM_SIGNAL:
+                    // In interactive mode, killed by a system signal is mostly likely by a user (e.g. Ctrl-C)
+                    return new JobProcessResult
+                        .Builder(JobStatus.KILLED,
+                        isInteractiveMode
+                        ? JobStatusMessages.JOB_KILLED_BY_USER : JobStatusMessages.JOB_KILLED_BY_SYSTEM,
```

Comment:
Generally for ternary operators we've gone with this kind of formatting if it is unable to fit on a single line:
```
isInteractiveMode
    ? JobStatusMessages.JOB_KILLED_BY_USER
    : JobStatusMessages.JOB_KILLED_BY_SYSTEM;
```